subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/action/InstrumentationOrientationChangeAction.kt

```
@@ -0,0 +1,29 @@
+package com.avito.android.test.action
+
+import android.app.Activity
+import android.content.pm.ActivityInfo
+import android.content.res.Configuration
+import androidx.test.platform.app.InstrumentationRegistry
+import com.avito.android.test.util.getCurrentActivity
+
+internal class InstrumentationOrientationChangeAction {
+
+    fun rotate() {
+        val instrumentation = InstrumentationRegistry.getInstrumentation()
+
+        instrumentation.runOnMainSync {
```

Could lead to timeouts when there is no Activity

---

Without Activity, it fails fastly with an error "No activities in RESUMED state found".
The behavior is the same as before. It also fails for this reason.